File: Cpp/1267_Count_Servers_that_Communicate.cpp
class Solution {
public:
    int countServers(vector<vector<int>>& grid) {
        int result = 0;
        int m = grid.size();
        int n = grid[0].size();
        vector<int> row(m, 0);
        vector<int> col(n, 0);
        
        for(int i = 0; i<m; ++i){
            for(int j = 0; j<n; ++j){
                if(grid[i][j] == 0) continue;
                ++row[i];
                ++col[j];
            }
        }
        
        for(int i = 0; i<m; ++i){
            for(int j = 0; j<n; ++j){
                if(grid[i][j] == 0
                  || (row[i]<2 && col[j]<2)) continue;
                result += 1;
            }
        }
        
        return result;
    }
};

File: Kotlin/1267_Count_Servers that_Communicate.kt
class Solution {
    fun countServers(grid: Array<IntArray>): Int {
        var result = 0
        val m = grid.size
        val n = grid[0].size
        val row = IntArray(m)
        val col = IntArray(n)
        
        for(i in 0 until m){
            for(j in 0 until n){
                if(grid[i][j] == 0) continue
                ++row[i]
                ++col[j]
            }
        }
        
        for(i in 0 until m){
            for(j in 0 until n){
                if(grid[i][j] == 0
                  || (row[i] < 2 && col[j] < 2)) continue
                ++result
            }
        }
        
        return result
    }
}